proposals/0000-improved-ghc-development.rst

```diff
@@ -0,0 +1,216 @@
+
+Proposed Change
+---------------
+- Link Phabricator and GitHub
+
+  Phabricator seems to have built in support for OAuth. One easy way so lower the barrier is to support OAuth logins to GitHub. This way a user doesn't have to create a new account. They can simply login using their GitHub credentials. *(This seems to already be enabled, maybe we should advertise it more.)*
+
+  Bonus points if the GitHub API allows us to extract the user's public key from GitHub. This would also mean less setup for the user to be able to push a diff.
+
+  Phabricator documentation for this https://secure.phabricator.com/book/phabricator/article/configuring_accounts_and_registration/
+  
+- Link Trac to GitHub
+
+  We already have a read-only mirror for code that is incredibly useful. I propose extending this read-only mode further. 
+
+   * Support OAuth on Trac for logins against GitHub as well.
+   * Synchronize Trac issues in a read-only way to GitHub. The usefulness of this feature comes In the ability for users to reference the issues from their own GitHub projects. This gives us some valuable information such as who these issues affect. This information would get synched back to Trac. This also allows users to be notified when something gets fixed. Since the issue would be closed on GitHub as well. Finally it gives some information on the importance of issues.
+
+  With the ability to login directly on Trac using your GitHub account, I believe it's easier to take issues on one place: Trac. So read-only would be enough.
+
+  As a whole, I find Trac MUCH better for ticket triaging than Github or Phab,
+  both of which seem to be quite bare and simple in what they provide. I am not
+  in favor of ditching it. Whatever we gain we lose in the ability to generate custom views on issues. Which is a powerful filtering mechanism for prioritization.
+
+  Also we have and continue to accept patches just uploaded
+  to Trac as a diff. We tend to ask people to upload it to phab for better reviews
+  and so it's attributed to them when we commit. Some don't (and we then do it ourselves),
+  most due. If they don't need another login then I suspect almost all would.
+
+  There's a (seemingly) actively maintained project that does all the above, could we leverage it?
+  https://github.com/trac-hacks/trac-github
+
+- Allow doc changes on Trac
+
+  I am not sure about the feasibility of this one, but there is a Trac plugin that adds a new section to trac under /doc that is able to render .rst files and more from a source repository. Essentially making a wiki from them.
+
+  https://github.com/trac-hacks/tracdocs
+
+  We could have harbormaster build the docs and provide them to trac. I am not sure how reviews would go. Styling would be another issue. 
+
+  Lastly this plugin seems to only support SVN. but that's not a huge barrier.
```

Comment:
It sounds like this could be a fair amount of work to get working but if someone was willing to put in the time then I'd be happy to see it done.

I suspect an easier way forward would be to simply have Harbormaster upload its documentation artifacts somewhere publicly accessible and post a link to them on the associated Differential. This would likely be just a few more lines to the build script.